composer.json

```diff
@@ -17,11 +17,17 @@
         "dhii/php-cs-fixer-config": "dev-php-5.3",
         "phpunit/phpunit": "^4.8",
         "ptrofimov/xpmock": "^1.1",
-        "codeclimate/php-test-reporter": "<=0.3.2"
+        "codeclimate/php-test-reporter": "<=0.3.2",
+        "dhii/validation-interface": "^0.1"
```

Comment:
> However the reference to ValidationFailedExceptionInterface should be enough to merit a non-dev dependency

I don't see how. Let's say, you have code which optionally handles a certain type, and does some extra normalization with it. Should you require the whole lib containing that type, just so that you can run the check?

WRT PHP 7, I've found nothing that says that the unavailability of the return type itself in the code would result in any kind of error or exception. As far as I know, this works in the same way as function arguments. Therefore, there is no need to require the interface package.